code/game/objects/items/devices/flashlight.dm

```diff
@@ -70,12 +70,13 @@
 			user.visible_message("<span class='notice'>[user] directs [src] to [M]'s eyes.</span>", \
 								 "<span class='notice'>You direct [src] to [M]'s eyes.</span>")
 
-			if(istype(M, /mob/living/carbon/human))	//robots and aliens are unaffected
-				if(M.stat == DEAD || M.disabilities & BLIND)	//mob is dead or fully blind
-					to_chat(user, "<span class='notice'>[M] pupils does not react to the light!</span>")
-				else if(XRAY in M.mutations)	//mob has X-RAY vision
-					to_chat(user, "<span class='notice'>[M] pupils give an eerie glow!</span>")
-				else	//they're okay!
+			if(istype(H)) //robots and aliens are unaffected
+				var/obj/item/organ/internal/eyes/eyes = H.get_int_organ(/obj/item/organ/internal/eyes)
+				if(M.stat == DEAD || !eyes || M.disabilities & BLIND)	//mob is dead or fully blind
+					to_chat(user, "<span class='notice'>[M]'s pupils are unresponsive to the light!</span>")
+				else if((XRAY in M.mutations) || (eyes.colourblind_darkview && eyes.colourblind_darkview == eyes.dark_view)) //The mob's either got the X-RAY vision or has a tapetum lucidum (extreme nightvision).

```

Comment:
Does this need to be updated to use `get_dark_view`, or is it only intended to happen for wolpins and farwas?